vendor/build-logger/src/ldlogger-tool-gcc.c

```diff
@@ -38,6 +38,20 @@ static const char* const srcExts[] = {
   "c", "cc", "cpp", "cxx", "h", "hh", "hxx", "hpp", "o", "so", "a", NULL
 };
 
+/**
+ * List of compiler name infixes belonging to C compilers.
+ */
+static const char* const cCompiler[] = {
```

Comment:
Why is this hardcoded? Shouldn't we use the GCC_LIKE and similar environment variables?

Reply:
Does clang support these environment variables?

Reply:
It does, see the environment section of https://clang.llvm.org/docs/CommandGuide/clang.html

Reply:
That means if we do not remove them the analyzers environment clang will use them during the analysis. Does clang-tidy support this too?

Reply:
This does not mean that. The environment of logging and the analysis might differ.

Reply:
Yes I know that, but if these environment variables are set before codechecker is run than the same same environment is used for analysis with some extensions. In that case the analysis environment would contain these variables.

Reply:
@gyorb: This pull request is about inserting the content of these environment variables to the build command in build.json. The analyzers will get this extended command, so it is not required that these variables are set in analysis phase.

@Xazax-hun: These are hard-coded because we have to distinguish C and C++ compilers. `CC_LOGGER_GCC_LIKE` contains these together. However I extend this list by the other elements of `CC_LOGGER_GCC_LIKE` to handle those too.

Reply:
In case the build system sets these variables, it might not be the same.
Build system might start a subshell that is used for compilation command, but the same subshell is not used for the analysis actions.

Reply:
@Xazax-hun I think we are speaking about completely different environment setups. I understand the cases you described and why this is needed on those cases.